### PR TITLE
Improve Diagnostic ordering further for multiple diagnostics with some of the same labels

### PR DESCRIPTION
**Pull request by brmataptos**
### Description

Improve ordering of Errors/Warnings shown to user by comparing more than one Label/location when there are multiple locations in two diagnostics.

Fixes https://github.com/aptos-labs/aptos-core/issues/10935

### Test Plan
Testing on existing move-compiler-v2 tests shows a change in 7 cases, manually checked they are an improvement.

### CODE REVIEW

File: third_party/move/move-model/src/model.rs [hunk 1]
```diff
@@ -1130,48 +1130,30 @@ impl GlobalEnv {
...
-                let second2 = labels2.iter().min_by(|l1, l2| GlobalEnv::cmp_label(l1, l2));
-                match (second1, second2) {
-                    (Some(sec1), Some(sec2)) => GlobalEnv::cmp_label(sec1, sec2),
-                    (Some(_), None) => Ordering::Less, // Label beats None
-                    (None, Some(_)) => Ordering::Greater, // None is beaten by Label
-                    (None, None) => Ordering::Equal,
-                }
-            },
-        }
+        let mut sorted_labels1: Vec<_> = labels1.iter().collect();
```
Comment (vineethk) [inline third_party/move/move-model/src/model.rs:1149]: nit: by using `collect_vec()`, you can be explicit about what you are collecting into (at the point of collection), as well as avoid the explicit type annotation.

**Review by vineethk — APPROVED** [review]
LGTM (only a minor comment).

File: third_party/move/move-model/src/model.rs (hunk 1 above)
Comment (brmataptos) [reply third_party/move/move-model/src/model.rs]: Done.

**Review by rahxephon89 — APPROVED** [review]
LGTM, thanks!

**Review by wrwg — APPROVED** [review]

**Pull request merged**